### ISSUE TITLE
Brainstorm some strategies to address student name collisions

### ISSUE THREAD

**Issue by zachmccormick** [TrailblazingTech/LSOA#25]
"students and name collisions.  We had some ideas about how to avoid this and to make it sensible to the teachers who are dealing with multiple Joe Smiths in and trying to figure out which is theirs…  But would also love to have your thinking…"

**Comment by cbradyatinquire** [comment]
For more on this, there are (at least) three issues.  One is Collisions; Two is Edits; Three is multi-year.

ONE
For Collisions:  If users are SYSTEM wide unique, we could have collisions between sites that are in different states (!), which would make it hard for teachers to resolve (or to identify in user listings who is who).
Right now, I think that users only have first and last names, so this is pretty likely to happen.  If we added school, then it would be easier to identify and disambiguate.  If we added an optional disambiguator for when collisions happened, this could help for user identification by teachers.  That disambiguator could even be an open text field.

TWO
It's probably important to avoid using the First or Last names as database keys, so that they are editable for things like  - spelling, changes to use nicknames or used-in-class names versus initially entered first names, and even fixes to last names.  For the last, for example: Corey Rodriguez Perez comes into class, and the school gives his last name as Perez, while he and his family use Rodriguez.

THREE
Multi year - when a student comes into the class, who's already in the system, the teacher needs to know which of possibly several students in the system is the one in front of them.  So the disambiguation solution has to work for identifying students already in the system.
